### PR TITLE
Enum dir rust/v1

### PR DESCRIPTION
**Pull request by inashivb**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/3832

### CODE REVIEW

**Comment by inashivb** [comment]
@jasonish @victorjulien do you like this?
Any changes I should make?
Should I do the same for other protocols as well?

**Comment by catenacyber** [comment]
Looks a good idea.
Can it be exported to C ? Like `SshStateFinished` for an example

**Comment by catenacyber** [comment]
(so as to have only one definition)

File: rust/src/core.rs [hunk 1]
```diff
@@ -39,6 +39,12 @@ pub const STREAM_GAP:      u8 = 0x10;
 pub const STREAM_DEPTH:    u8 = 0x20;
 pub const STREAM_MIDSTREAM:u8 = 0x40;
 
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub enum Direction {
+    ToServer = 0x04,
+    ToClient = 0x08,
+ }
+
```
Comment (jasonish) [inline rust/src/core.rs:47]: Can you try adding repr(C) here as well?  Then maybe it can be the source of truth and used from C as well.

**Review by jasonish — CHANGES_REQUESTED** [review]
I'd like to see this rebased.  Its a good candidate to become an enum on the Rust side I think.

**Comment by inashivb** [comment]
Replaced w #6294

**Pull request closed without merging**